File: server/src/main/java/com/fluffy/oauth2/application/OAuth2Strategy.java
package com.fluffy.oauth2.application;

import com.fluffy.auth.domain.OAuth2Provider;
import com.fluffy.oauth2.domain.OAuth2UserInfo;

public interface OAuth2Strategy {

    String buildOAuth2LoginUrl(String next);

    OAuth2UserInfo fetchOAuth2UserInfo(String code);

    String buildClientRedirectUrl(String next);

    OAuth2Provider getProvider();
}

File: server/src/main/kotlin/com/fluffy/oauth2/application/OAuth2Strategy.kt
package com.fluffy.oauth2.application

import com.fluffy.auth.domain.OAuth2Provider
import com.fluffy.oauth2.domain.OAuth2UserInfo

interface OAuth2Strategy {

    fun buildOAuth2LoginUrl(next: String): String

    fun fetchOAuth2UserInfo(code: String): OAuth2UserInfo

    fun buildClientRedirectUrl(next: String): String

    fun getProvider(): OAuth2Provider
}